packages/astro/src/core/env/config.ts

```diff
@@ -0,0 +1,48 @@
+const fieldsConstants = {
+	optional: true,
+	staticScope: 'static',
+	dynamicScope: 'dynamic',
+	publicAccess: 'public',
+	privateAccess: 'public',
+	stringType: 'string',
+} as const;
+
+const createFields = <
+	TScope extends (typeof fieldsConstants)['staticScope' | 'dynamicScope'],
+	TAccess extends (typeof fieldsConstants)['publicAccess' | 'privateAccess'],
+>(
+	scope: TScope,
+	access: TAccess
+) => {
+	const shared = { scope, access };
+	return {
+		string: () => ({
+			...shared,
+			type: fieldsConstants.stringType,
+			optional: () => ({
+				...shared,
+				type: fieldsConstants.stringType,
+				optional: fieldsConstants.optional,
+				default: (value: string) => ({
+					...shared,
+					type: fieldsConstants.stringType,
+					optional: fieldsConstants.optional,
+					default: value,
+				}),
+			}),
+		}),
+	};
+};
+
+/**
+ * TODO:
+ */
+export const envField = {
```

Comment:
Types don't match actually because of the methods but I'm waiting until we're a bit more sure about what to support and how to name things